test/test_client.py

```diff
@@ -1948,6 +1948,8 @@ def test_exhaust_getmore_network_error(self):
         self.assertRaises(ConnectionFailure, list, cursor)
         self.assertTrue(conn.closed)
 
+        # Wait for the deferred killCursor to complete
+        time.sleep(1)
```

Comment:
Instead of a sleep it would be more robust to wait_until the client's `__kill_cursors_queue` is empty.